### PR TITLE
JetBrains preview and open in editor

### PR DESCRIPTION
**Pull request by vdavid**
Fixes https://github.com/sourcegraph/sourcegraph/issues/34368

## Test plan

The features are tested and demonstrated here: https://www.loom.com/share/63c6e922db514ac0a65c1ef7e3561a23

The code is not live and it's a WIP, so I'd say **we only need a stamp here**.

### CODE REVIEW

**Review by taras-yemets — APPROVED** [review]

File: client/jetbrains/webview/src/search/results/FileSearchResult.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ export const FileSearchResult: React.FunctionComponent<Props> = ({
                 <div className={styles.lineCode}>
                     <TrimmedCodeLineWithHighlights line={line} />
                 </div>
-                <div className={classNames(styles.lineLineNumber, 'text-muted')}>{line.lineNumber}</div>
+                <div className={classNames(styles.lineLineNumber, 'text-muted')}>{line.lineNumber + 1}</div>
```
Comment (philipp-spiess) [inline client/jetbrains/webview/src/search/results/FileSearchResult.tsx:49]: 👀

File: client/jetbrains/webview/src/search/index.tsx [hunk 2]
```diff
@@ -22,16 +17,19 @@ declare global {
     }
 }
 
-async function onOpen(match: ContentMatch, lineIndex: number): Promise<void> {
-    console.log('open', await loadContent(match), match.lineMatches[lineIndex])
+async function onPreviewChange(match: ContentMatch, lineMatchIndex: number): Promise<void> {
+    await window.callJava(await createRequestForMatch(match, lineMatchIndex, 'preview'))
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/jetbrains/webview/src/search/index.tsx:21]: Nice, would be cool, I've added a task for myself to "Move `onPreviewChange()` and similar functions to `jsToJavaBridgeUtil.ts` to hide `window.callJava()` from the rest of the app."

File: client/jetbrains/webview/src/search/index.tsx [hunk 3]
```diff
@@ -22,16 +17,19 @@ declare global {
...
 }
 
-async function onPreviewChange(match: ContentMatch, lineIndex: number): Promise<void> {
-    console.log('preview', await loadContent(match), match.lineMatches[lineIndex])
+function onPreviewClear(): void {
+    window
+        .callJava({ action: 'clearPreview', arguments: {} })
+        .then(() => {})
+        .catch(() => {})
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/jetbrains/webview/src/search/index.tsx:28]: I agree, added that to the same task as well. I'll create a common GitHub issue for a collection of housekeeping tasks.

File: client/jetbrains/webview/src/search/jsToJavaBridgeUtil.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            content,
+            lineNumber: match.lineMatches[lineMatchIndex].lineNumber,
+            absoluteOffsetAndLengths,
+        },
+    }
+}
+
+function getCharacterCountUntilLine(content: string, lineNumber: number): number {
+    let count = 0
+    const lines = content.split('\n') // This logic should handle \r\n well, too.
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/jetbrains/webview/src/search/jsToJavaBridgeUtil.ts:38]: Yeah, I'll add a comment about this to the code

File: client/jetbrains/webview/src/search/results/FileSearchResult.tsx (hunk 1 above)
Comment (vdavid) [reply client/jetbrains/webview/src/search/results/FileSearchResult.tsx:49]: Numbering is hard :D

**Pull request merged**